app/models.py

```diff
@@ -497,6 +498,38 @@ def generate_oauth_client_id(client_name) -> str:
     return generate_oauth_client_id(client_name)
 
 
+class MfaBrowser(db.Model, ModelMixin):
+    user_id = db.Column(db.ForeignKey(User.id, ondelete="cascade"), nullable=False)
+    token = db.Column(db.String(64), default=False, nullable=False)
```

Comment:
should the `token` be `unique`?

Reply:
Yes but for a 64 character lowercase string (26^64 possibilities), this is **very** unlikely. Like a [1 in 196 duovigintillion chance](https://www.wolframalpha.com/input/?i=1%C3%B7%28%281%C3%B7%2826%5E64%29%29%C3%972%5E64%29) for a collision even when you generate 2^64 tokens.

Reply:
I agree :) but just **in theory** this can happen right? The `random_string` is also pseudorandom so in a very unlikely case, someone could affect the random generation, putting the data in an inconsistent state.

Reply:
Haha yeah absolutely true ;)